src/schema/schema_template.py

```diff
@@ -42,51 +51,127 @@ async def query_schema_templates(
         query = filter_model_query(models_template.SchemaTemplate, filters, query=query)
     schema_templates = (await session.exec(query)).all()
 
-    return [
-        GraphQLSchemaTemplate(id=template.id, name=template.name, schema=schema)
-        for template, schema in schema_templates
-    ]
+    for template, schema in schema_templates:
+        query = select(models_category.SchemaCategory).where(
+            models_category.SchemaCategory.reporting_schema_id == schema.id
+        )
+        schema_categories = (await session.exec(query)).all()
 
+        joined_schema = schema_reporting.GraphQLReportingSchema(
+            id=schema.id,
+            name=schema.name,
+            categories=schema_categories,
+            project_id=schema.project_id,
+            template_id=template.id,
+        )
+        template = GraphQLSchemaTemplate(id=template.id, name=template.name, schema=joined_schema)
+        templates.append(template)
 
-async def add_schema_template_mutation(info: Info, name: str) -> GraphQLSchemaTemplate:
-    """Add a Schema Template"""
+    return templates
 
+
+async def add_schema_template_mutation(
+    info: Info, name: str, type_codes: Optional[list[GraphQLTypeCodeElementInput]] = None
+) -> GraphQLSchemaTemplate:
+    """Add a Schema Template"""
     session = info.context.get("session")
     schema_template = models_template.SchemaTemplate(name=name)
     session.add(schema_template)
 
+    reporting_schema = models_reporting.ReportingSchema(name=name, template=schema_template)
+    session.add(reporting_schema)
+
+    if type_codes:
+        for type_code in type_codes:
+            schema_category = models_category.SchemaCategory(
+                name=type_code.name, path=type_code.code, depth=type_code.level, reporting_schema=reporting_schema
```

Comment:
Depth doesn't exist on the `SchemaCategory` model, because it is inferred from the path. See the `depth` method of the `GraphQLSchemaCategory` class.

The path is constructed a bit like an Unix folder structure with `/` to separate each level.
Instead of using folder names as you do in a folder structure, we use the id of the parents in the path.
See picture below.

![image](https://github.com/lcacollect/documentation-backend/assets/16247318/b14fd99a-941a-4a82-9182-5ada499f0d1e)

The top level category (1 | Bygningsbasis) has a path `/`, giving it depth 0.
The child (10 | Terræn) has the path `"/765e776c-7d1a-4caa-9137-4b05982ef9b2"`, which is a `/` + the id of the parent. That gives it the depth of 1.
The "grandchild" (10x | Terræn) has the path `"/765e776c-7d1a-4caa-9137-4b05982ef9b2/ffd02ded-d1b2-489c-9248-e6b7333cf54c"`, which is the ids of its parent and grandparent separated with slashes. That gives the depth of 2.

Therefore we need to capture the structure/hierarchy from the "Create New Template" form.

![Screenshot from 2023-10-02 09-58-18_2](https://github.com/lcacollect/documentation-backend/assets/16247318/63ad3555-861d-4748-9170-2f0c8e2e8e9a)